### PR TITLE
feat: Safely auto-resolve stale project-state contradictions (#1636)

### LINKED ISSUE #1636: Safely auto-resolve stale project-state contradictions
## Summary

Hybrid-memory maintenance successfully runs `resolve-contradictions`, but the current auto-resolution policy leaves a large number of safe project-state contradictions unresolved. In the 2026-05-27 manual maintenance sweep, `resolve-contradictions` exited successfully but reported:

```text
Contradictions resolved: 0 auto-resolved, 223 ambiguous.
```

A follow-up analysis of those 223 ambiguous pairs showed this is mostly not genuine semantic conflict. It is primarily mutable project/task state (`status`, `next`, `task_updated`, etc.) being stored as competing current facts instead of superseding previous state.

The goal of this issue is to make contradiction cleanup more autonomous while preserving safety for non-project facts.

## Evidence from 2026-05-27 maintenance sweep

Sweep log:

```text
/home/markus/.openclaw/logs/cron-hybrid-mem/nightly-memory-sweep-20260527T034314Z-30772.log
```

Exit summary:

```text
/home/markus/.openclaw/logs/cron-hybrid-mem/nightly-memory-sweep-20260527T034314Z-30772.exit.txt
```

Relevant successful maintenance outcomes:

```text
2026-05-27T03:43:49Z prune exit=0
2026-05-27T04:35:48Z distill exit=0
2026-05-27T04:35:58Z extract-daily exit=0
2026-05-27T04:36:08Z resolve-contradictions exit=0
2026-05-27T05:16:24Z enrich-entities exit=0
```

Quality improvements observed:

```text
Pruned 309 expired facts. Before: 56924, After: 56615. Vector cleanup: 309/309 deleted.
Distill done: 667 stored, 204 skipped (888 extracted from 1352 sessions).
Entity enrichment: processed 200 facts, enriched 57 (batch had 200 candidates).
Contradictions resolved: 0 auto-resolved, 223 ambiguous.
```

## Analysis of ambiguous contradictions

Manual analysis of the 223 ambiguous pairs found:

- `223/223` had the newer fact on the left side (`fact_id_new` was indeed newer).
- `223/223` were blocked only because the newer fact confidence was **not strictly greater** than the older fact's original confidence.
- `218/223` were `project ↔ project` category pairs.
- `204/223` were likely stale project state keys:
  - `status`: 106
  - `next`: 90
  - `task_updated`: 8
- `145/223` involved `active-task` generated facts.
- The dominant source pairs were:
  - `conversation ↔ active-task`: 145
  - `conversation ↔ conversation`: 76
  - `conversation ↔ distillation`: 2

Top recurring `entity/key` clusters:

```text
  9 entity='hybrid-memory' key='next'
  7 entity='hybrid-memory PR queue' key='next'
  7 entity='hybrid-memory-open-issue-coverage-2026-05-22' key='coverage'
  5 entity='hybrid-memory-pr-processing-oldest-first' key='next'
  5 entity='hybrid-memory-issue-1272-pr' key='next'
  5 entity='hybrid-memory-open-issues-to-prs' key='status'
  5 entity='goal-heartbeat-reliability-hardening' key='next'
```

These examples are mostly state transitions, not true contradictions. For example:

```text
hybrid-memory-issue-1271-pr / status
older: in_progress — PR #1282 had CI/Biome failures
newer: done — PR #1282 merged
recommendation: supersede older current-state fact, retain provenance
```

```text
hybrid-memory-open-issue-coverage-2026-05-22 / coverage
older: 27_open_26_covered_1_uncovered
newer: 27_open_27_covered_0_uncovered
recommendation: supersede older coverage snapshot for current-state retrieval
```

## Current behavior/root cause

`resolveContradictionsAuto()` currently only supersedes when all of these are true:

- new fact exists and old fact exists
- new fact is newer
- new fact confidence is strictly greater than old original confidence
- new fact source is `conversation` or `cli`

For project-state facts, equal confidence is common and expected. This makes safe state transitions remain ambiguous forever.

In the sampled run, every ambiguous pair had `old_fact_original_confidence = 1.0`; many newer facts were also `1.0`, so the strict `>` confidence condition prevented resolution.

### 1. Add a project-state latest-wins resolver policy

For mutable project state, allow a newer trusted fact to supersede an older fact when confidence is equal or higher.

Candidate safe keys:

```text
status
next
task_updated
related_session
coverage
last_live_verified_at
live_state_hash
last_actionable_blocker
owner
```

Suggested policy:

```text
If category = project
and same entity + key + scope/scope_target
and key is in projectStateKeys
and newer.created_at > older.created_at
and newer.source is trusted for project state
and newer.confidence >= old_original_confidence
then supersede old with new and resolve contradiction as superseded.
```

Trusted sources for this policy could include:

```text
conversation
cli
active-task
```

Optionally include `distillation` only when confidence is higher or when explicitly configured.

### 2. Keep the generic resolver conservative

Do **not** loosen contradiction resolution globally. Personal facts, credentials, technical truths, emails, and non-project facts should continue to require stricter evidence or manual review.

This should be a targeted policy for project/task state only.

### 3. Prevent future active-task contradiction churn

`active-task`/project ledger writes should preferably supersede the previous fact for the same `entity + key + scope`, instead of creating contradiction records.

Options:

- write project-state facts with explicit `supersedes` where possible
- add a helper for project fact upserts that marks the previous current fact superseded
- exclude same-key project-state active-task snapshots from contradiction creation if they are just mutable state

Recommendation: implement canonical project fact upsert/supersession at write time, so cleanup is not only retrospective.

### 4. Add an operator-visible dry-run/apply command

Add a safe command such as:

```bash
openclaw hybrid-mem resolve-contradictions --project-state-lww --dry-run
openclaw hybrid-mem resolve-contradictions --project-state-lww --apply
```

Dry-run output should group by `entity + key`, not just UUIDs.

Example output:

```text
project-state-lww candidates: 204
would supersede: 201
manual review: 3

hybrid-memory PR queue / next
  keep: 2026-05-23 12:12 conversation conf=1.0 "No pending work — PR #1605 is CLEAN..."
  supersede: 2026-05-23 09:44 conversation conf=0.8 "PR #1605: 2 non-outdated threads remain..."
```

### 5. Improve contradiction reporting

Current output is UUID-heavy and difficult to act on. Add a `--details`/`--grouped` format with:

- entity
- key
- fact ids
- timestamps
- source
- confidence
- value/text excerpt
- suggested action:
  - `auto-safe: project-state-lww`
  - `manual: non-project`
  - `manual: overloaded entity/follow-up task`
  - `ignore/history-only`

### 6. Detect overloaded task entities

Some task entities appear reused for follow-up/sibling work. Example pattern:

```text
hybrid-memory-issue-1272-pr:
- PR #1284 done
- later follow-up PR #1288 in progress
- later PR #1288 done
```

This is not exactly an old-wrong/new-right conflict; it is a task entity being reused for a new phase. The resolver can still supersede for current-state retrieval, but reporting should flag these as possible entity hygiene issues.

Possible heuristic:

- same entity/key has multiple PR numbers or issue numbers across values/text
- status goes `done -> in_progress -> done`
- entity name contains an issue/PR number but text references different PR/issue numbers

Suggested recommendation in report: split future follow-up work into a new entity or add a phase/subtask key.

## Acceptance criteria

- [ ] `resolve-contradictions` supports a project-state latest-wins policy that safely resolves stale `project` facts for known mutable keys.
- [ ] The policy only applies to same `entity + key + scope/scope_target` and trusted sources.
- [ ] Equal-confidence newer project-state facts can supersede older facts under this policy.
- [ ] Generic/non-project contradiction behavior remains conservative.
- [ ] A dry-run mode reports candidate counts and grouped details before applying changes.
- [ ] An apply mode marks contradictions resolved and supersedes stale facts with provenance preserved.
- [ ] Active-task/project ledger writes avoid creating avoidable contradictions for mutable current-state keys.
- [ ] Reporting groups contradictions by `entity + key` and includes human-readable values/timestamps/sources/confidence.
- [ ] Tests cover:
  - equal-confidence project `status` transition `in_progress -> done`
  - project `next` update from `active-task -> conversation`
  - non-project equal-confidence conflict remains ambiguous
  - overloaded task entity is flagged or at least not silently hidden in reports
  - dry-run does not mutate facts/contradictions
  - apply mode resolves expected contradictions and supersedes old facts

## Safety notes

This should not become a broad “newer always wins” rule. The safe autonomy comes from limiting the rule to mutable project/task state keys and preserving provenance. For user identity, personal data, emails, technical truths, credentials, and general facts, ambiguity should still require manual review.

## Operator impact

Expected impact based on the 2026-05-27 sweep: this would likely reduce the ambiguous contradiction queue from 223 to a small manual-review set, while improving current-state retrieval for active tasks and project stewardship.

### PR DESCRIPTION
**Pull request by markus-lassfolk**
Implements issue #1636: adds a targeted **project-state latest-wins (LWW)** policy that safely auto-resolves stale project/task-state contradictions for known mutable keys, while keeping all other contradiction handling conservative and unchanged.

## Overview

The existing `resolveContradictionsAuto` path left many ambiguous contradiction pairs unresolved because equal-confidence project-state facts (status, next, task_updated, etc.) didn't meet the strict "strictly higher confidence" requirement. This PR adds a separate, scoped LWW policy that handles those cases safely.

### Core logic (`backends/facts-db/contradictions.ts`)
- **`PROJECT_STATE_LWW_KEYS`** — 9 mutable project/task state keys eligible for LWW (`status`, `next`, `task_updated`, `related_session`, `coverage`, `last_live_verified_at`, `live_state_hash`, `last_actionable_blocker`, `owner`). Documented to explain why these are "current-state snapshot" keys (as opposed to append-style facts like decisions or milestones)
- **`PROJECT_STATE_LWW_TRUSTED_SOURCES`** — `conversation`, `cli`, `active-task`; distillation is explicitly excluded to keep automated facts conservative
- **`evaluateLwwEligibility()`** — exported shared helper used by both the write-time path and the batch pass to keep eligibility logic in a single place
- **`MAX_EXPECTED_REFS_PER_ENTITY`** — named constant for the overloaded-entity heuristic threshold (replaces inline magic number)
- **`resolveProjectStateLww()`** — batch resolution pass that evaluates all unresolved contradictions and emits grouped `supersede`/`manual-review` candidates; dry-run safe

### Safety guards
Every pair must satisfy **all** of:
- Both facts have `category = "project"`
- Key (case-insensitive) is in `PROJECT_STATE_LWW_KEYS`
- Newer fact's source is in `PROJECT_STATE_LWW_TRUSTED_SOURCES`
- Newer fact is strictly newer (`createdAt >` older fact)
- Newer fact confidence ≥ older fact's original confidence (captured at detection time, before the −0.2 penalty)

Pairs with many distinct PR/issue refs across their values are flagged as `possibleOverloadedEntity` in the report without blocking resolution.

### FactsDB write-time path (`backends/facts-db/facts-db-layer3.ts`)
- `detectContradictions()` now immediately resolves qualifying project-state pairs at write time using `evaluateLwwEligibility`, so active-task/project writes no longer leave avoidable unresolved contradictions
- New `resolveContradictionsProjectStateLww({ dryRun? })` method for the batch/nightly pass

### CLI (`cli/commands/manage/register-reflection-pipeline.ts`)
- `--project-state-lww` flag on `resolve-contradictions` to run the LWW batch pass
- `--dry-run` flag (with `--project-state-lww`) to report candidates with grouped details without mutating any data

### Service wiring
New method plumbed through `cli-services.ts` → `register-help.ts` → `cli/context.ts` → `cli/register.ts`.

## Tests

New test file `tests/project-state-lww.test.ts` with 9 tests covering all acceptance criteria:
- Equal-confidence `status` transition (`in_progress` → `done`)
- `next` key update across sources (`active-task` → `conversation`)
- Non-project equal-confidence conflict stays ambiguous (unchanged behavior)
- Overloaded entity flagging
- Dry-run immutability (no mutations)
- Distillation source guard (not eligible for LWW)
- Older-wins case → `manual-review` action
- Write-time auto-supersede on `detectContradictions`

All 373 test files (6390 tests) pass.

Fixes #1636

---

> [!NOTE]
> **Medium Risk**
> Changes when facts are auto-superseded and how contradictions resolve at write time and via CLI; mistakes could drop or mis-rank project memory, though guards (trusted sources, verified facts, missing original confidence) limit blast radius.
> 
> **Overview**
> Adds a **project-state latest-wins (LWW)** path for contradiction handling so stale `project` facts on mutable keys (`status`, `next`, etc.) can be superseded when a newer trusted fact wins, without changing the conservative default for other categories.
> 
> **Facts / contradiction engine:** `recordContradiction` / `detectContradictions` now return **pre-penalty** `oldFactOriginalConfidence`. Conservative auto-resolve is refactored behind shared `getAutoResolutionDecision`, with a non-mutating **`previewResolveContradictionsAuto`**, stricter apply behavior (supersede failures stay ambiguous; verified facts block auto-supersede unless the old row is already superseded). New **`evaluateLwwEligibility`**, **`resolveProjectStateLww`**, and write-time LWW in **`FactsDB.detectContradictions`** for eligible keys/sources.
> 
> **CLI:** `resolve-contradictions` gains **`--dry-run` / `--apply`** (and **`--project-state-lww`**) for grouped LWW preview/apply output; default apply mode unchanged. Wiring adds `previewResolveContradictions` and `resolveContradictionsProjectStateLww` through CLI context.
> 
> **Self-correction (smaller):** LLM JSON parsing accepts **`NO_ACTION`** without `remediationContent`; repair pass uses the same parser; `tryParseFirstJsonArray` supports an optional filter callback.
> 
> **Tests:** New coverage for project-state LWW, CLI contract, verified+superseded preview/apply parity, and NO_ACTION parsing.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 891d87e107b72fe5689226cf6db7ab037542b8f3. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by markus-lassfolk** [comment]
@copilot implement this issue on the current PR branch.

Repo/Issue: markus-lassfolk/openclaw-hybrid-memory#1636 https://github.com/markus-lassfolk/openclaw-hybrid-memory/issues/1636
PR: #1641 https://github.com/markus-lassfolk/openclaw-hybrid-memory/pull/1641
Branch: issue-1636-safely-auto-resolve-stale-project-state-contradi
Head: 233b0a0f87f175bcfe777cc896349ad9b1ec71bc
Draft: true
Labels: size/XS,stage/implementing,priority:high

Implementation brief:
Use the issue and PR body as the implementation brief. If acceptance is unclear, report the exact blocker instead of broadening scope.

Constraints:
- Push to this existing PR branch; do not open a separate PR.
- Keep scope focused on issue #1636: Safely auto-resolve stale project-state contradictions
- Preserve current behavior outside the acceptance criteria.
- Add/update tests and docs as described in the enrichment.
- If acceptance is unclear or blocked, comment with the exact blocker.

next_owner=Copilot; wait_until=Copilot push/comment or timeout.

**Comment by markus-lassfolk** [comment]
@copilot implement this issue on the current PR branch.

Repo/Issue: markus-lassfolk/openclaw-hybrid-memory#1636 https://github.com/markus-lassfolk/openclaw-hybrid-memory/issues/1636
PR: #1641 https://github.com/markus-lassfolk/openclaw-hybrid-memory/pull/1641
Branch: issue-1636-safely-auto-resolve-stale-project-state-contradi
Head: 07f6b121fa78e628b287ff4d84886c63b005eff7
Draft: true
Labels: extensions,size/XL,stage/implementing,priority:high

Implementation brief:
Use the issue and PR body as the implementation brief. If acceptance is unclear, report the exact blocker instead of broadening scope.

Constraints:
- Push to this existing PR branch; do not open a separate PR.
- Keep scope focused on issue #1636: Safely auto-resolve stale project-state contradictions
- Preserve current behavior outside the acceptance criteria.
- Add/update tests and docs as described in the enrichment.
- If acceptance is unclear or blocked, comment with the exact blocker.

next_owner=Copilot; wait_until=Copilot push/comment or timeout.

**Comment by markus-lassfolk** [comment]
@copilot please fix this bounded blocker on the current PR head.

Repo/PR: markus-lassfolk/openclaw-hybrid-memory#1641
Head: 07f6b121fa78e628b287ff4d84886c63b005eff7
Blocker: Format Check (Biome) FAILING — https://github.com/markus-lassfolk/openclaw-hybrid-memory/actions/runs/26496409404/job/78025606595
Expected fix: Run `biome format --write` on the changed files and push to the existing branch `issue-1636-safely-auto-resolve-stale-project-state-contradi`
Verification: Format Check (Biome) passes on next CI run

Stewardship next_owner=Copilot; attempt=1/2; will recheck after CI settles.

**Comment by markus-lassfolk** [comment]
@copilot please repair this bounded blocker on the current PR head and push to the existing branch.

Repo/PR: markus-lassfolk/openclaw-hybrid-memory#1641 https://github.com/markus-lassfolk/openclaw-hybrid-memory/pull/1641
Title: feat: Safely auto-resolve stale project-state contradictions (#1636)
Head SHA: dab149a7fb07be982a08fe090ee55c8e43d65028
Branch: issue-1636-safely-auto-resolve-stale-project-state-contradi
Current merge state: MERGEABLE/BEHIND
Labels: extensions,size/XL,stage/council,priority:high
Blocker category: unknown
Blocker signature: 730062c4cd3a (dirty-conflict-or-ci)
Attempt: 1/2 for this blocker lineage

Blockers:
- MERGEABLE/BEHIND (unknown)

Requested work:
- Fix the listed blocker(s) on this PR head.
- Push to the existing PR branch; do not open a new PR.
- Preserve this PR's feature intent and current main behavior.
- Report what was fixed, commit SHA, verification run, and any remaining blockers.

next_owner=Copilot; escalation_tier=copilot1; wait_until=Copilot push/comment or timeout.

**Comment by markus-lassfolk** [comment]
@copilot please repair this bounded blocker on the current PR head and push to the existing branch.

Repo/PR: markus-lassfolk/openclaw-hybrid-memory#1641 https://github.com/markus-lassfolk/openclaw-hybrid-memory/pull/1641
Title: feat: Safely auto-resolve stale project-state contradictions (#1636)
Head SHA: 2f5900bb9034544e7ea27d30ddc58cba4676cb54
Branch: issue-1636-safely-auto-resolve-stale-project-state-contradi
Current merge state: MERGEABLE/BEHIND
Labels: extensions,size/XL,fixes-needed,stage/review,priority:high
Blocker category: fixes-needed
Blocker signature: 10061d719b92 (fixes-needed)
Attempt: 1/2 for this blocker lineage

Blockers:
- MERGEABLE/BEHIND (fixes-needed)

Requested work:
- Fix the listed blocker(s) on this PR head.
- Push to the existing PR branch; do not open a new PR.
- Preserve this PR's feature intent and current main behavior.
- Report what was fixed, commit SHA, verification run, and any remaining blockers.

next_owner=Copilot; escalation_tier=copilot1; wait_until=Copilot push/comment or timeout.

File: extensions/memory-hybrid/backends/facts-db/contradictions.ts [hunk 1]
```diff
@@ -192,6 +199,75 @@ export function getContradictedIds(db: DatabaseSync, factIds: string[]): Set<str
...
+      ambiguous.push({
+        contradictionId: c.id,
+        factIdNew: c.factIdNew,
+        factIdOld: c.factIdOld,
+      });
+    }
+  }
+
+  return { autoResolvable, ambiguous };
+}
```
[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply extensions/memory-hybrid/backends/facts-db/contradictions.ts:305]: Fixed in d55986367918bdd702b5a63140410fa99b68ae01 by removing the preview/apply divergence in classic contradiction auto-resolution.

What changed:
- Added shared helper `isAutoResolvableContradiction(...)` in `extensions/memory-hybrid/backends/facts-db/contradictions.ts`.
- `previewResolveContradictionsAuto` and `resolveContradictionsAuto` now both use that helper.
- Helper keeps existing apply semantics: if the old fact is already superseded, the contradiction is auto-resolvable even when the old fact is verified.

Regression coverage:
- Added `keeps preview/apply consistent when old fact is verified but already superseded` in `extensions/memory-hybrid/tests/contradiction-detection.test.ts`.
- Ran:
  - `npm exec -- biome lint --max-diagnostics=none backends/facts-db/contradictions.ts tests/contradiction-detection.test.ts`
  - `npm test -- tests/contradiction-detection.test.ts tests/project-state-lww.test.ts`
- Result: pass (2 files, 40 tests).

Reference thread: https://github.com/markus-lassfolk/openclaw-hybrid-memory/pull/1641#discussion_r3316966871

**Comment by markus-lassfolk** [comment]
Fixed: previewResolveContradictionsAuto now reuses the shared eligibility check from resolveContradictionsAuto.

File: extensions/memory-hybrid/cli/cmd-selfcorrection.ts [hunk 2]
```diff
@@ -54,6 +59,88 @@ const DEFAULT_SELF_CORRECTION = {
...
+    )
+  ) {
+    return false;
+  }
+
+  if ("category" in candidate && typeof candidate.category !== "string") return false;
+  if ("severity" in candidate && typeof candidate.severity !== "string") return false;
+  if ("repeated" in candidate && typeof candidate.repeated !== "boolean") return false;
+
+  return true;
```
[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply extensions/memory-hybrid/cli/cmd-selfcorrection.ts:135]: Fixed on `05b9716c`.

Resolution:
- Updated `isSelfCorrectionRemediationItem` to accept `remediationType: "NO_ACTION"` when `remediationContent` is omitted.
- Kept strict validation for actionable remediation types (`remediationContent` is still required unless type is `NO_ACTION`).
- Added parser regressions in `tests/self-correction-run-parser.test.ts` for both accepted and rejected cases.

Validation run:
- `npm --prefix extensions/memory-hybrid test -- tests/self-correction-run-parser.test.ts` (20 tests passed)
- `npm exec -- biome lint --max-diagnostics=none cli/cmd-selfcorrection.ts tests/self-correction-run-parser.test.ts`

[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply extensions/memory-hybrid/cli/cmd-selfcorrection.ts:135]: Fixed on current head.

What changed:
- `isSelfCorrectionRemediationItem` now allows missing `remediationContent` only for `NO_ACTION` (`remediationType` normalized via `trim().toUpperCase()`), while still requiring valid string/object content for actionable remediation types.
- Added focused regressions:
  - parser accepts `NO_ACTION` items without `remediationContent`
  - parser still rejects actionable items missing `remediationContent`
  - run-level self-correction test verifies `NO_ACTION` without content is treated as no-op (no parse error).

Validation run:
- `npm test -- tests/self-correction-run-parser.test.ts tests/reinforcement-analysis.test.ts` (pass: 38 tests)

Commits on this branch:
- `05b9716c` (code-path fix in parser validator)
- `cbc04341` (focused regression tests)

**Comment by markus-lassfolk** [comment]
@copilot repair

This PR has merge conflicts that must be resolved before it can advance. The `fixes-needed` label is set — resolve the conflicts and push the updated branch.

PR: https://github.com/markus-lassfolk/openclaw-hybrid-memory/pull/1641
Branch: `issue-1636-safely-auto-resolve-stale-project-state-contradi`
Current head: `91a783339d1f478c572b8cf10998eba6b3503460`
Required action: rebase on latest main, resolve conflicts, push to the same branch.

**Comment by markus-lassfolk** [comment]
Promoting current head `a0ef3653cf909d1159e2b0effc7c3c07e2e8a663` to `stage/council`: current-head checks are green/acceptable, review threads are resolved, no blocker labels are present, and merge state is not dirty/conflicting/behind. Next owner: Council/deep review dispatch.

**Comment by markus-lassfolk** [comment]
Council/deep verification request marker for current head (audit breadcrumb only).

Repo/PR: markus-lassfolk/openclaw-hybrid-memory#1641 https://github.com/markus-lassfolk/openclaw-hybrid-memory/pull/1641
Title: feat: Safely auto-resolve stale project-state contradictions (#1636)
Head SHA: 59bf4c639305a66fe3f142d1d65eaf0ec93a132c
Current state: merge_state=MERGEABLE/CLEAN; labels=extensions,size/XL,stage/council,priority:high; non-Council merge gates passed with Council gate disabled.
Dispatch bridge: COUNCIL_SESSION_BRIEF emitted at /tmp/council-session-brief-markus-lassfolk_openclaw-hybrid-memory-pr1641-59bf4c639305.json
Required Council output (parseable):
- COUNCIL_REVIEW_HEAD: 59bf4c639305a66fe3f142d1d65eaf0ec93a132c
- COUNCIL_VERDICT: APPROVED | CHANGES_REQUESTED
- If CHANGES_REQUESTED: REQUIRED checklist items with stable REQUIRED:<item_id>.

next_owner=Stewardship cron/main (sessions_spawn bridge); wait_until=Council verdict for current head.

**Comment by markus-lassfolk** [comment]
Council review for PR #1641 head 59bf4c639305a66fe3f142d1d65eaf0ec93a132c.

COUNCIL_REVIEW_HEAD: 59bf4c639305a66fe3f142d1d65eaf0ec93a132c
COUNCIL_VERDICT: CHANGES_REQUESTED

COUNCIL_REQUIRED_ITEMS:
- [ ] REQUIRED:item_1 Wire the new project-state LWW resolver into the operator apply contract requested by #1636. The issue/enrichment explicitly require `resolve-contradictions --dry-run` / `resolve-contradictions --apply` semantics and apply-mode dry-run parity; current head adds only `--dry-run` and `--project-state-lww`, with apply as an implicit “omit --dry-run” mode. `resolve-contradictions --apply` is not registered, and the existing default `resolve-contradictions` path still runs only the conservative resolver, so the motivating maintenance sweep would not automatically clean the stale project-state backlog unless an operator knows to run a new non-acceptance command form. Add the explicit apply flag/contract (or otherwise make the documented acceptance command work), reject ambiguous flag combinations, and cover it with CLI-level tests. owner=forge severity=high
- [ ] REQUIRED:item_2 Emit the structured/auditable summary line required for cron/operator monitoring, including the requested `auto-resolved=X dry-run=Y remaining=Z` style fields in both dry-run and apply paths. Current output is human-readable (`project-state-lww candidates...` / `project-state-lww (applied)...`) but does not provide the required structured fields for detecting zero-cleanup or over-cleanup anomalies. Add tests that assert the CLI output includes grouped entity+key details and the structured summary in empty, dry-run, and apply cases. owner=forge severity=medium

**Comment by markus-lassfolk** [comment]
- [x] REQUIRED:item_1 status=fixed evidence=commit:b6a93aa7 files:extensions/memory-hybrid/cli/commands/manage/register-reflection-pipeline.ts tests:tests/resolve-contradictions-cli.test.ts::supports --dry-run contract mode;supports --apply contract mode;rejects ambiguous flag combinations
- [x] REQUIRED:item_2 status=fixed evidence=commit:b6a93aa7 files:extensions/memory-hybrid/cli/commands/manage/register-reflection-pipeline.ts tests:tests/resolve-contradictions-cli.test.ts::supports --dry-run contract mode;supports --apply contract mode;prints structured summary in empty dry-run output

**Comment by markus-lassfolk** [comment]
Rebased onto current main to resolve merge conflict.

**Comment by markus-lassfolk** [comment]
Promoting current head `891d87e107b72fe5689226cf6db7ab037542b8f3` to `stage/council`: current-head checks are green/acceptable, review threads are resolved, no blocker labels are present, and merge state is not dirty/conflicting/behind. Next owner: Council/deep review dispatch.

**Comment by markus-lassfolk** [comment]
Council/deep verification request marker for current head (audit breadcrumb only).

Repo/PR: markus-lassfolk/openclaw-hybrid-memory#1641 https://github.com/markus-lassfolk/openclaw-hybrid-memory/pull/1641
Title: feat: Safely auto-resolve stale project-state contradictions (#1636)
Head SHA: 891d87e107b72fe5689226cf6db7ab037542b8f3
Current state: merge_state=MERGEABLE/CLEAN; labels=extensions,size/XL,stage/council,priority:high; non-Council merge gates passed with Council gate disabled.
Dispatch bridge: COUNCIL_SESSION_BRIEF emitted at /tmp/council-session-brief-markus-lassfolk_openclaw-hybrid-memory-pr1641-891d87e107b7.json
Required Council output (parseable):
- COUNCIL_REVIEW_HEAD: 891d87e107b72fe5689226cf6db7ab037542b8f3
- COUNCIL_VERDICT: APPROVED | CHANGES_REQUESTED
- If CHANGES_REQUESTED: REQUIRED checklist items with stable REQUIRED:<item_id>.

next_owner=Stewardship cron/main (sessions_spawn bridge); wait_until=Council verdict for current head.

**Comment by markus-lassfolk** [comment]
COUNCIL_REVIEW_HEAD: 891d87e107b72fe5689226cf6db7ab037542b8f3
COUNCIL_VERDICT: APPROVED
